static/app/views/replays/replays.tsx

```diff
@@ -34,34 +36,59 @@ type Props = AsyncView['props'] &
     statsPeriod?: string | undefined; // revisit i'm sure i'm doing statsperiod wrong
   };
 
-class Replays extends React.Component<Props> {
-  getEventView() {
-    const {location, selection} = this.props;
-    const {query} = location;
+const sanitizeLocationQuery = query => {
+  if (!query || Array.isArray(query)) {
+    return '';
+  }
+  return query;
+};
+
+function Replays(props: Props) {
+  const location = useLocation();
+  const organization = useOrganization();
+  const {projects} = useProjects();
+
+  const [searchQuery, setSearchQuery] = useState<string>(
+    sanitizeLocationQuery(location.query.query)
+  );
+
+  useEffect(() => {
+    setSearchQuery(sanitizeLocationQuery(location.query.query));
+  }, [location.query.query]);
 
+  const getEventView = () => {
+    const {selection} = props;
+    const {query} = location;
     const eventQueryParams: NewQuery = {
       id: '',
       name: '',
       version: 2,
       fields: ['eventID', 'project', 'timestamp', 'user.display', 'url'],
+      // @ts-ignore
       orderby: query.sort || '-timestamp',
       environment: selection.environments,
       projects: selection.projects,
-      query: 'transaction:sentry-replay', // future: change to replay event
+      query: `transaction:sentry-replay ${searchQuery}`, // future: change to replay event
     };
 
     if (selection.datetime.period) {
       eventQueryParams.range = selection.datetime.period;
     }
     return EventView.fromNewQueryWithLocation(eventQueryParams, location);
-  }
+  };
 
-  getTitle() {
-    return `Replays - ${this.props.params.orgId}`;
-  }
+  const handleSearchQuery = (query: string) => {
```

Comment:
We absolutely could, though I'm tempted to say not at the moment since I doubt this would cause such a performance hit to justify memoizing the function right now. I'm good with doing it if there's a strong inclination to do so 👍